vercel/turbo
# Wire filewatching to daemon server

### Description

 - daemon asynchronously loads filewatching
 - filewatching errors and/or repo root being removed trigger server shutdown
 - filewatcher constructor is now fully asynchronous
 - running gRPC server is now a function call, rather than a struct to simplify ownership and lifecycle of filewatching-related tasks.

### Testing Instructions

Existing tests, plus an additional test at the server level for root deletion.

To be done before re-enabling rust daemon:
 - make an intelligent choice of cookie directory
 - verify timeout behavior so we don't block the client ever

Closes TURBO-1317
